controllers/openstackdataplanerole_controller.go

```diff
@@ -106,28 +122,105 @@ func (r *OpenStackDataPlaneRoleReconciler) Reconcile(ctx context.Context, req ct
 	}
 
-	if len(nodes.Items) == 0 {
-		r.Log.Info("Role: ", instance.Name, "doesn't have nodes yet, requeueing")
-		return ctrl.Result{RequeueAfter: time.Second * 5}, nil
-	}
-
 	for _, node := range nodes.Items {
 		if node.Spec.Role != instance.Name {
 			err = fmt.Errorf("node %s: node.Role does not match with node.Label", node.Name)
 			return ctrl.Result{}, err
 		}
-		if !node.Status.Deployed {
-			r.Log.Info("Node: ", node.Name, "isn't deployed yet, requeueing")
-			return ctrl.Result{RequeueAfter: time.Second * 2}, nil
-		}
 	}
 
-	_, err = r.GenerateInventory(ctx, instance, nodes.Items)
+	inventoryConfigMap, err := r.GenerateInventory(ctx, instance, nodes.Items)
 	if err != nil {
 		util.LogErrorForObject(helper, err, fmt.Sprintf("Unable to generate inventory for %s", instance.Name), instance)
 		return ctrl.Result{}, err
 	}
 
+	// Always patch the instance status when exiting this function so we can
+	// persist any changes.
+	defer func() {
+		err := helper.PatchInstance(ctx, instance)
+		if err != nil {
+			r.Log.Error(_err, "PatchInstance error")
+			_err = err
+			return
+		}
+	}()
+
+	ansibleSSHPrivateKeySecret := instance.Spec.NodeTemplate.AnsibleSSHPrivateKeySecret
+	_, result, err = secret.VerifySecret(
+		ctx,
+		types.NamespacedName{Namespace: instance.Namespace, Name: ansibleSSHPrivateKeySecret},
+		[]string{
+			"ssh-privatekey",
+		},
+		helper.GetClient(),
+		time.Duration(5)*time.Second,
+	)
+	if err != nil {
+		return result, err
+	}
+
+	// Initialize Status
+	if instance.Status.Conditions == nil {
+		instance.Status.Conditions = condition.Conditions{}
+
+		cl := condition.CreateList(
+			condition.UnknownCondition(dataplanev1beta1.DataPlaneRoleReadyCondition, condition.InitReason, condition.InitReason),
+			condition.UnknownCondition(dataplanev1beta1.ConfigureNetworkReadyCondition, condition.InitReason, condition.InitReason),
+			condition.UnknownCondition(dataplanev1beta1.ValidateNetworkReadyCondition, condition.InitReason, condition.InitReason),
+			condition.UnknownCondition(dataplanev1beta1.InstallOSReadyCondition, condition.InitReason, condition.InitReason),
+			condition.UnknownCondition(dataplanev1beta1.ConfigureOSReadyCondition, condition.InitReason, condition.InitReason),
+			condition.UnknownCondition(dataplanev1beta1.RunOSReadyCondition, condition.InitReason, condition.InitReason),
+			condition.UnknownCondition(dataplanev1beta1.InstallOpenStackReadyCondition, condition.InitReason, condition.InitReason),
+			condition.UnknownCondition(dataplanev1beta1.ConfigureOpenStackReadyCondition, condition.InitReason, condition.InitReason),
+			condition.UnknownCondition(dataplanev1beta1.RunOpenStackReadyCondition, condition.InitReason, condition.InitReason),
+		)
+
+		instance.Status.Conditions.Init(&cl)
+
+		instance.Status.Deployed = false
+
+		// Register overall status immediately to have an early feedback e.g.
+		// in the cli
+		return ctrl.Result{}, nil
+	}
+
+	if instance.Spec.DeployStrategy.Deploy {
```

Comment:
shouldn't we set 
```go
instance.Status.Deployed = true
```
at the end of this if?

Reply:
This is set in deployment.go so the logic is shared between nodes and roles.